src/target/riscv/riscv-013.c

```diff
@@ -1023,6 +1033,23 @@ static int read_abstract_arg(struct target *target, riscv_reg_t *value,
 	return result;
 }
 
+static void abstract_data_write_fill_batch(struct riscv_batch *batch,
+		riscv_reg_t value, unsigned int index, unsigned int size_bits)
+{
+	assert(size_bits % 32 == 0);
+	const unsigned int size_in_words = size_bits / 32;
+	const unsigned int offset = index * size_in_words;
+
+	for (unsigned int i = 0; i < size_in_words; ++i) {
+		const unsigned int reg_address = DM_DATA0 + offset + i;
+
+		riscv_batch_add_dm_write(batch, reg_address, value,
```

Comment:
Here I'd recommend to explicitly truncate the `value` - for code readers' benefit:

```suggestion
		riscv_batch_add_dm_write(batch, reg_address, (uint32_t)value,
```

Reply:
Addressed.